### DISCUSSION TITLE
Duplicate Share Issue RTM

### DISCUSSION THREAD

**Discussion by Jimmy062006** [coinfoundry/miningcore#1301] [General]
**Describe the bug**
When mining the miner will report for RTM. I think possibly because the same job id is used?

**To Reproduce**
1. Launch XMRig
2. Mine to the pool
3. Duplicate shares will appear

**Expected behavior**
No duplicate shares

**Additional context**
`{"jsonrpc":"2.0","method":"mining.notify","params":["00002e87","d32a951c72b8e768392d82cfc9f7dd73b8e1c3c2a990271dbc867e9213f3c2a7","03000500010000000000000000000000000000000000000000000000000000000000000000ffffffff1d03cd050404d548256200","0a4d696e696e67636f72650000000003a8ea7648170000001976a9143542a8dd417574b43e091fb246b6ca46b6a54b7288acaaba1dd2050000001976a914cc36a8de36f6ff61dbfb798982e8b47354ad4e0e88acf8efbd4f570000001976a9149fdf79eacbfae060f9344cbcc1864e0b2f3a579c88ac00000000460200cd0504001aae02b30746cdb0a86e0626b50f042c80eec359293f416533f8af37dac35b41822e50a1386107de3180098c5965de207e9063a83a1e32a821eca94d9f421f22",["01e793ff27660b201792f9ab16c3a43d7704e3d1fcf90940a9376a2d31adc123","b272feacb6b5dabfa0292388ad943b7ac3547338d3a90240fee03ad02ea4d7f0","03aabbeec95d753a3494d4e0216b6578d6791bf4ce71a4a2db5984ba2e21edb7"],"20000000","1c38040a","622548d5",false],"id":null}
{"id":99,"jsonrpc":"2.0","method":"mining.submit","params":["RRAWxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.WORKER001","00002e87","00000000","622548d5","001a8010"]}
{"result":true,"id":99}
{"jsonrpc":"2.0","method":"mining.set_difficulty","params":[1.0869940759657157],"id":null}
{"jsonrpc":"2.0","method":"mining.notify","params":["00002e87","d32a951c72b8e768392d82cfc9f7dd73b8e1c3c2a990271dbc867e9213f3c2a7","03000500010000000000000000000000000000000000000000000000000000000000000000ffffffff1d03cd050404d548256200","0a4d696e696e67636f72650000000003a8ea7648170000001976a9143542a8dd417574b43e091fb246b6ca46b6a54b7288acaaba1dd2050000001976a914cc36a8de36f6ff61dbfb798982e8b47354ad4e0e88acf8efbd4f570000001976a9149fdf79eacbfae060f9344cbcc1864e0b2f3a579c88ac00000000460200cd0504001aae02b30746cdb0a86e0626b50f042c80eec359293f416533f8af37dac35b41822e50a1386107de3180098c5965de207e9063a83a1e32a821eca94d9f421f22",["01e793ff27660b201792f9ab16c3a43d7704e3d1fcf90940a9376a2d31adc123","b272feacb6b5dabfa0292388ad943b7ac3547338d3a90240fee03ad02ea4d7f0","03aabbeec95d753a3494d4e0216b6578d6791bf4ce71a4a2db5984ba2e21edb7"],"20000000","1c38040a","622548d5",false],"id":null}
{"id":100,"jsonrpc":"2.0","method":"mining.submit","params":["RRAWxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.WORKER001","00002e87","00000000","622548d5","001a8010"]}
{"result":false,"error":{"code":22,"message":"duplicate share","data":null},"id":100}`

**Comment by Jimmy062006** [comment]
Hi Any progress on this bug ?

**Comment by Jimmy062006** [comment]
This is still an issue. Any suggestion on where to look to diagnose this issue a little better?

**Comment by Jimmy062006** [comment]
Just to point out this still exists  and if there is anything else needed to continue looking in to this.

**Reply by fer000n** [reply]
same issue on ghostrider

**Comment by Censseo** [comment]
same here, we have more than 1Mh/s on miningcore, and a lot a duplicate share. There is another issue with this by the way, this is the huge amounts of job sent, especially when the algo changes and switch to faster ones. I think we should dig this way, maybe higher the amount of nonce in jobs. Didn't dig into the code yet to see how it is working.
